wandb/apis/public/api.py

```diff
@@ -1624,8 +1636,276 @@ def slack_integrations(
+
+            automation = api.create_automation(
+                triggering_event >> triggered_action,
+                name="my-automation",
+                description="Send a Slack message whenever 'custom-metric' exceeds 10.",
+            )
+            ```
+        """
+        from wandb.automations import ActionType, Automation
+        from wandb.automations._generated import (
+            CREATE_FILTER_TRIGGER_GQL,
+            CreateFilterTrigger,
+            NoOpActionFields,
+        )
+        from wandb.automations._utils import prepare_to_create
+
+        gql_input = prepare_to_create(obj, **kwargs)
+
+        if not self._supports_automation(
+            event=(event := gql_input.triggering_event_type),
+            action=(action := gql_input.triggered_action_type),
+        ):
+            raise ValueError(
+                f"Automation event or action ({event!r} -> {action!r}) "
+                "is not supported on this wandb server version. "
+                "Please upgrade your server version, or contact support at "
+                "support@wandb.com."
+            )
+
+        # keep the name on hand, if needed later to report on
+        # and/or recover from a conflicting automation
+        name = gql_input.name
+
+        # If needed, rewrite the GraphQL field selection set to omit unsupported fields/fragments/types
+        omit_fragments = set()
+        if not self._supports_automation(action=ActionType.NO_OP):
+            omit_fragments.add(NoOpActionFields.__name__)
+
+        mutation = gql_compat(CREATE_FILTER_TRIGGER_GQL, omit_fragments=omit_fragments)
+        variables = {"params": gql_input.model_dump(exclude_none=True)}
+
+        try:
+            data = self.client.execute(mutation, variable_values=variables)
+        except requests.HTTPError as e:
+            status = HTTPStatus(e.response.status_code)
+            if status is HTTPStatus.CONFLICT:  # 409
+                if fetch_existing:
+                    wandb.termlog(f"Automation {name!r} exists. Fetching it instead.")
+                    return self.automation(name=name)
+
+                raise ValueError(
+                    f"Automation {name!r} exists. Unable to create another with the same name."
+                )
+            raise e
+
+        try:
+            result = CreateFilterTrigger.model_validate(data).result
+        except ValidationError as e:
+            msg = f"Invalid response while creating automation {name!r}"
+            raise RuntimeError(msg) from e
+
+        if (result is None) or (result.trigger is None):
+            msg = f"Empty response while creating automation {name!r}"
+            raise RuntimeError(msg)
+
+        return Automation.model_validate(result.trigger)
+
+    def delete_automation(self, obj: Union[str, "Automation"]) -> Literal[True]:
+        """Delete an automation.
+
+        Args:
+            obj (str | Automation): The automation to delete, or its ID.
```

Comment:
```suggestion
            obj (str | Automation): The ID of the Automation to delete or its Automation object.
```
Is this what you mean? It seems to match the order you present the choices here.

Reply:
Eh I'll switch the union order to maintain brevity here, but side note for later -- it would be preferable if our generated docs automatically parsed + documented the the type hints directly from the function signature.  This'll keep our docs more accurate + avoid the need to manually list arg types in the docstring